mlflow/utils/autologging_utils.py

```diff
@@ -46,13 +46,14 @@ def get_unspecified_default_args(user_args, user_kwargs, all_param_names, all_de
             if name not in user_specified_arg_names}
 
 
-def log_fn_args_as_params(fn, args, kwargs, unlogged=[]):  # pylint: disable=W0102
+def log_fn_args_as_params(fn, args, kwargs, unlogged=[], unpacked=[]):  # pylint: disable=W0102
     """
     Log parameters explicitly passed to a function.
     :param fn: function whose parameters are to be logged
     :param args: arguments explicitly passed into fn
     :param kwargs: kwargs explicitly passed into fn
     :param unlogged: parameters not to be logged
+    :param unpacked: dict parameters to be unpacked
```

Comment:
Maybe clarify this comment, seems like the idea of this parameter is to allow special handling of function arguments that correspond to dictionaries of params (e.g. `{"learning_rate": 0.01"}`)? Maybe we should just exclude such params via `unlogged` & log them separately for now, since AFAICT the XGBoost autologging functionality is the only one with this use case at the moment?

Reply:
Agree with excluding dict params via `unlogged` and logging them separately. I'll revert this change.